app/1.0.x/plugin-development/admin-api.md

```diff
@@ -34,37 +34,80 @@ Kong will detect and load your endpoints if they are defined in a module named:
+  passed to a function (by default form fields are all strings).
+- `method1` and `method2` are strings representing the usual HTTP methods: `GET`, `POST`, etc.
+   The special strings `before` and `on_error` can also be used (described below)
+- `function1` and `function2` describe what to do when a particular path-method combination is matched
+  (or for the special `before` function). They can be Lua functions, but often the `kong.api.endpoints` module provides some default
+  implementation that is appropiate.
+
+For example:
+
+``` lua
+local endpoints = require "kong.api.endpoints"
 
-This table is then fed to Lapis (See Lapis' [handling HTTP verbs
-documentation](http://leafo.net/lapis/reference/actions.html#handling-http-verbs)).
-Example:
+local credentials_schema = kong.db.keyauth_credentials.schema
+local consumers_schema = kong.db.consumers.schema
 
-```lua
 return {
-  ["/my-plugin/new/get/endpoint"] = {
-    GET = function(self, dao_factory, helpers)
-      -- ...
-    end
-  }
+  ["/consumers/:consumers/key-auth"] = {
+    schema = credentials_schema,
+    methods = {
+      GET = endpoints.get_collection_endpoint(
+              credentials_schema, consumers_schema, "consumer"),
+
+      POST = endpoints.post_collection_endpoint(
+              credentials_schema, consumers_schema, "consumer"),
+    },
+  },
 }
 ```
 
-The handler function takes three arguments, which are, in order:
+This code will create two Admin API endpoints in `/consumers/:consumers/key-auth`, to
+obtain (`GET`) and create (`POST`) credentials associated to a given consumer. On this example
+the functions are provided by the `kong.api.endpoints` library. If you want to see a more
+complete example, with custom code in functions, see
+[the real keyauth `api.lua` file](https://github.com/Kong/kong/blob/master/kong/plugins/key-auth/api.lua).
+
+The `endpoints` module currently contains the default implementation for the most usual CRUD
+operations used in Kong. This module provides you with helpers for any insert, retrieve,
+update or delete operations and performs the necessary DAO operations and replies with
+the appropriate HTTP status codes. It also provides you with functions to retrieve parameters from
+the path, such as an API's name or id, or a Consumer's username or id.
+
+For those cases where the provided default functions are not enough, you can write your own
+handler functions in Lua as well. They take three arguments, which are, in order:
 
 - `self`: The request object. See [Lapis request
   object](http://leafo.net/lapis/reference/actions.html#request-object)
-- `dao_factory`: The DAO Factory. See the
-  [datastore]({{page.book.chapters.access-the-datastore}}) chapter of this
-  guide.
-- `helpers`: A table containing a few helpers, described below.
+- `db`: A shortcut to `kong.db`. It was mainly used before Kong 1.0
+- `helpers`: A table containing a few helpers. Currently superseeded by the Kong PDK.
```

Comment:
superseeded -> superseded

BUT... if these two arguments are deprecated, could we just not document them anymore?

Reply:
yep, removed those params